engine/include/Engine/Core/Parsers/ShaderParser.hpp

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2021 Amara Sami, Dallard Thomas, Nardone William, Six Jonathan
+ * This file is subject to the LGNU license terms in the LICENSE file
+ *	found in the top-level directory of this distribution.
+ */
+
+#pragma once
+
+#include <string>
+
+namespace GPE
+{
+// singletone
+class ShaderParser
+{
+public:
+    ShaderParser()                          = delete;
+    ShaderParser(const ShaderParser& other) = delete;
+    ShaderParser(ShaderParser&& other)      = delete;
+    ~ShaderParser()                         = delete;
+
+    /**
+     * @brief call and use all parse function
+     *
+     * @param sourceCode
+     */
+    static inline void parse(std::string& sourceCode)
+    {
+        parseIncludesShader(sourceCode);
+    }
```

Comment:
Why not directly call `parseIncludesShader()`? Removing this function and renaming `parseIncludesShader()` to `parse()` would work too if it feels better that way

Reply:
Because it can include more parse tools. For now, I just have implemented a parsing include tool. In the future, we will probabely add more shader parsing tools

Reply:
I would call `parseIncludesShader()` to something along the lines of `includeHeaders()` (since `Shader` is already in the class' name)